ios/brave-ios/Sources/BraveShared/Reachability.swift

```diff
@@ -22,14 +22,31 @@ public enum ReachabilityType: CustomStringConvertible {
     case .offline: return "Offline"
     }
   }
+
+  fileprivate init(path: NWPath) {
+    let type: ReachabilityType
+    if path.status == .satisfied {
+      if path.usesInterfaceType(.wifi) {
+        type = .wifi
+      } else if path.usesInterfaceType(.cellular) {
+        type = .cellular
+      } else if path.usesInterfaceType(.wiredEthernet) {
+        type = .ethernet
+      } else {
+        type = .other
+      }
+    } else {
+      type = .offline
+    }
+    self = type
+  }
 }
 
-open class Reachability: ObservableObject {
+public class Reachability: Observable {
```

Comment:
This is not how you use `Observable`, you need to use the `@Observable` macro

Reply:
Fixed. Used `@Observable` and `withObservationTracking` recursively.